### DISCUSSION TITLE
Can we make the starlight layout a part of the website, not an entity website?

### DISCUSSION THREAD

**Discussion by sgalcheung** [withastro/starlight#2389] [Feature Requests]
Since I started an experiment for pulling doc content from 3rd-party API ([from discussion #1790](https://github.com/withastro/starlight/discussions/1790#discussioncomment-10658211)), I started exploring writing a single Astro page, but this can't share common <html/> page.
Also, I noted in other discussions, that so far, to custom page is radical(come from https://github.com/withastro/starlight/discussions/2108#discussioncomment-9981160).

**Comment by delucis** [comment]
If I understand correctly, you want your Starlight pages to be just a part of your Astro site. That’s supported! We have a few notes in the documentation here: https://starlight.astro.build/manual-setup/#tips-for-existing-sites

But in general, yes, you can use your custom pages using content from an API alongside regular Starlight pages 🙌

**Reply by sgalcheung** [reply]
My experiment project([starlight-squidex](https://github.com/sgalcheung/starlight-squidex)) has already been implemented from API render contents.
There is another puzzle is how Starlight documents as a subpath of the Astro page. As you mentioned link(Tips for existing sites), it is a static document pulled from the content folder, since I pulled it from API, I haven't files in the content folder.

**Reply by delucis** [reply]
Ah right. Then you have all the flexibility of [Astro routing](https://docs.astro.build/en/guides/routing/). You can either create subdirectories in `src/pages/` to nest things or if you’re using the [`injectRoute()`](https://docs.astro.build/en/reference/integrations-reference/#injectroute-option) integration API then similarly you an adjust the pattern.

**Reply by sgalcheung** [reply]
Yep, that's exactly what I did, then I tried to integrate Tailwind CSS, and I recognized that Starlight has a layout Page for HTML, and my Astro pages in pages have another, so this is my problem.

**Reply by sgalcheung** [reply]
Another puzzle is that I don't know how I can inject data for the sidebar, so I added the squidex package to fetch data, it should inject from the enable example.
The sidebar also needs a dynamic change in the data source, now, this only executes once when starting the project.

**Reply by delucis** [reply]
In case you haven’t seen it yet https://github.com/HiDeoo/starlight-typedoc could be a helpful reference for the sidebar stuff.

**Reply by sgalcheung** [reply]
That is not suitable for my situation, the starlight-typedoc is injected at the build setup, and I need to render it at runtime because it is not the default situation, it depends on the user's actions.

**Reply by sgalcheung** [reply]
Just pass in the parameter sidebar saved me, like this
`<StarlightPage frontmatter={frontmatter} sidebar={sidebar}>`

**Reply by sgalcheung** [reply]
This is a dynamic load starlight, but the HTML layout changes the main website.

**Comment by sgalcheung** [comment]
This problem is still here.

The website at https://starlight-squidex.netlify.app
![image](https://github.com/user-attachments/assets/2486616f-4ea4-4b58-8fe9-1fc84e66ce41)


My dynamic load starlight at https://starlight-squidex.netlify.app/column/article/jianguo-hotel-qianmen
![image](https://github.com/user-attachments/assets/51ecd7c3-290c-4279-b1eb-621aedc47adb)
